mitodl/ocw-studio
# Update to Python 3.12.5

### What are the relevant tickets?
N/A; Python 3.12.5 was released on August 6, 2024.

### Description (What does it do?)
This PR updates the Python version used from 3.12.2 to 3.12.5. Other dependencies are updated as well.

### How can this be tested?
The testing instructions for https://github.com/mitodl/ocw-studio/pull/2141 should be followed. Specifically, rebuild containers with `docker compose build --no-cache`, and smoke test various aspects of the entire site.
